compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -84,10 +84,9 @@ internal class BindingModuleGenerator(
 
   // Keeps track of for which scopes which files were generated. Usually there is only one file,
   // but technically there can be multiple.
-  private val mergedScopes = mutableMapOf<FqName, MutableList<Pair<File, KtClassOrObject>>>()
-    .withDefault { mutableListOf() }
+  private val mergedClasses = mutableMapOf<MergedClassKey, File>()
 
-  private val excludedTypesForScope = mutableMapOf<FqName, List<FqName>>()
+  private val excludedTypesForClass = mutableMapOf<MergedClassKey, List<FqName>>()
```

Comment:
The bug is here where we collected all excluded types per scope rather than class with a merge annotation.